src/Lykke.Service.KucoinAdapter.Services/OrderBookProcessPipelines.cs

```diff
@@ -1,15 +1,20 @@
 ﻿using System;
 using System.Reactive.Linq;
 using Common.Log;
-using Lykke.Service.KucoinAdapter.Core.Domain.SharedContracts;
+using Lykke.Common.ExchangeAdapter.Contracts;
 
 namespace Lykke.Service.KucoinAdapter.Services
 {
     public static class OrderBookProcessPipelines
     {
+        private static (bool, string, OrderBook) FromTryGetPattern(OrderBook orderBook)
+        {
+            return (orderBook.TryDetectNegativeSpread(out var error), error, orderBook);
```

Comment:
Is this correct that an uninitialized variable  (error) is sent as an argument?

Reply:
It's initialized for sure because it comes as `out` argument, not `ref`. Personally, I don't like such notation, I prefer either `Result<T>` or `Nullable<T>`, but it's common in .NET (F# for example has native transparent patten-matching sugar from `bool TryGet<T>(out T val)` to `(bool, Option 't)`.